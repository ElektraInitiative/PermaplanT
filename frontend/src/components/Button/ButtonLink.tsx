import { MouseEventHandler } from 'react';
import { Link } from 'react-router-dom';

interface ButtonLinkProps {
  /** The title of the button. */
  title: string;
  /** The path to navigate to. */
  to: string;
  /** The className styling of the button. */
  className?: string;
  /** Clickhandler for the button. */
  onClick?: MouseEventHandler<HTMLButtonElement>;
}

/**
 * A styled link that has properties of a button.
 */
const ButtonLink = ({ title, to, className = '', onClick }: ButtonLinkProps) => {
  return (
    <Link
      to={to}
      className={`flex items-center ${className} justify-center space-x-2
                  hover:text-primary-500      active:text-primary-300
                  dark:hover:text-primary-500 dark:active:text-primary-300`}
    >
      <button onClick={onClick}>{title}</button>
    </Link>
  );
};

export default ButtonLink;
