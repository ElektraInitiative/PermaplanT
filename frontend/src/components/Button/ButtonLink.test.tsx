import ButtonLink from './ButtonLink';
import { MemoryRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';

it('renders correctly', () => {
  const tree = renderer
    .create(
      <MemoryRouter>
        <ButtonLink to="https://dev.permaplant.net" title="Permaplant" />
      </MemoryRouter>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
