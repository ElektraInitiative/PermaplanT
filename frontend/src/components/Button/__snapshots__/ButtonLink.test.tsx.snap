// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<a
  className="flex items-center  justify-center space-x-2
                  hover:text-primary-500      active:text-primary-300
                  dark:hover:text-primary-500 dark:active:text-primary-300"
  href="https://dev.permaplant.net"
>
  <button>
    Permaplant
  </button>
</a>
`;
